internal/commands/user/shared/inputs.go

```
@@ -0,0 +1,102 @@
+		if err != nil {
+			return nil, err
+		}
+		for _, user := range foundUsers {
+			selectableUsersSet[user.ID] = user
+		}
+	}
+	if i.Status == StatusTypePending || i.Status == StatusTypeNil {
+		userFilter.Pending = true
+		foundUsers, err := client.FindUsers(app.GroupID, app.ID, userFilter)
+		if err != nil {
+			return nil, err
+		}
+		for _, user := range foundUsers {
+			selectableUsersSet[user.ID] = user
+		}
+	}
+
+	// Interactive User Selection
+	selectableUsers := map[string]string{}
+	selectableUserOptions := make([]string, len(selectableUsersSet))
+	userOptionPattern := "%s - %s"
+	userOptIndex := 0
+	for _, user := range selectableUsersSet {
+		var opt string
+		switch user.Identities[0].ProviderType {
+		case ProviderTypeAnonymous:
+			opt = fmt.Sprintf(userOptionPattern, "Anonymous", user.ID)
+		case ProviderTypeLocalUserPass:
+			opt = fmt.Sprintf(userOptionPattern, user.Data["email"], user.ID)
+		case ProviderTypeAPIKey:
+			opt = fmt.Sprintf(userOptionPattern, user.Data["name"], user.ID)
+		case ProviderTypeApple:
+			opt = fmt.Sprintf(userOptionPattern, "Apple", user.ID)
+		case ProviderTypeGoogle:
+			opt = fmt.Sprintf(userOptionPattern, "Google", user.ID)
+		case ProviderTypeFacebook:
+			opt = fmt.Sprintf(userOptionPattern, "Facebook", user.ID)
+		case ProviderTypeCustom:
+			opt = fmt.Sprintf(userOptionPattern, "Custom JWT", user.ID)
+		case ProviderTypeCustomFunction:
+			opt = fmt.Sprintf(userOptionPattern, "Custom Function", user.ID)
+		}
+		selectableUserOptions[userOptIndex] = opt
+		selectableUsers[opt] = user.ID
+		userOptIndex++
+	}
+	selectedUsers := []string{}
+	err = ui.AskOne(
+		&selectedUsers,
+		&survey.MultiSelect{
+			Message: "Which user(s) would you like to delete?",
+			Options: selectableUserOptions,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	users := []string{}
+	for _, userOption := range selectedUsers {
+		users = append(users, selectableUsers[userOption])
+	}
+	return users, nil
```

[q] Should we return an error here if we found no users (or even handle that upstream in the command after this function returns)?

---

I personally think it should be handled slightly upstream. With potentially a feedback message saying "no users to delete".